test/end2end_test.go

```diff
@@ -6994,3 +6995,99 @@ func testLargeTimeout(t *testing.T, e env) {
 		}
 	}
 }
+
+// Proxies typically send GO_AWAY followed by connection closure a minute or so later. This
+// test ensures that the connection is re-created after GO_AWAY and not affected by the
+// subsequent (old) connection closure.
+func TestGoAwayThenClose(t *testing.T) {
+	defer leakcheck.Check(t)
+
+	ctx := context.Background()
+
+	lis1, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening. Err: %v", err)
+	}
+	defer lis1.Close()
```

Comment:
`s1.Stop()` will take care of this.  No need to do it separately.

Reply:
Done